src/Analyzers/CSharp/CodeFixes/ConvertTypeOfToNameOf/CSharpConvertTypeOfToNameOfCodeFixProvider.cs

```diff
@@ -0,0 +1,91 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CSharpConvertTypeOfToNameOfCodeFixProvider
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharp.CSharpConvertTypeOfToNameOfCodeFixProvider)), Shared]
+    internal partial class CSharpConvertTypeOfToNameOfCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpConvertTypeOfToNameOfCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+           => ImmutableArray.Create(IDEDiagnosticIds.ConvertTypeOfToNameOfDiagnosticId);
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+               context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = (MemberAccessExpressionSyntax)editor.OriginalRoot.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                if (node != null && node is MemberAccessExpressionSyntax)
+                {
+                    ConvertTypeOfToNameOf(editor, node, semanticModel);
+                }
+            }
+        }
+
+        /// <Summary>
+        ///  Method converts typeof(...).Name to nameof(...)
+        /// </Summary>
+        internal static void ConvertTypeOfToNameOf(
+            SyntaxEditor editor, MemberAccessExpressionSyntax node, SemanticModel? semanticModel)
+        {
+            var typeOfExpression = (TypeOfExpressionSyntax)node.Expression;
+            var symbolType = semanticModel.GetSymbolInfo(typeOfExpression.Type).Symbol.GetSymbolType();
+
+            if (symbolType != null)
```

Comment:
this would be bad, as it would mean we offered a fix on something we couldn't actually fix.  we should check this up front in the analyzer to make sure we won't offer anythin gthat will later fail.